File: aoj/volume100/csharp/A10033.cs
using System;
using System.Collections.Generic;

class AOJ
{
	static void Main()
	{
		var noStack = int.Parse(Console.ReadLine());

		Stack<string>[] stacks = new Stack<string>[noStack];
		for (var i = 0; i < noStack; i++)
			stacks[i] = new Stack<string>();

		while (true)
		{
			var instruction = Console.ReadLine().Split();
			var operation = instruction[0];
			if (operation == "quit")
				break;

			int src, dst;
			string block;
			switch (operation)
			{
				case "push":
					dst = int.Parse(instruction[1])-1;
					block = instruction[2];
					stacks[dst].Push(block);
					break;
				case "pop":
					src = int.Parse(instruction[1])-1;
					block = stacks[src].Pop();
					Console.WriteLine(block);
					break;
				case "move":
					src = int.Parse(instruction[1])-1;
					dst = int.Parse(instruction[2])-1;
					stacks[dst].Push(stacks[src].Pop());
					break;
			}
		}
	}
}

File: aoj/volume100/scala/A10033.scala
object Main {
  def main(args: Array[String]) {
    val n_stacks = scala.io.StdIn.readInt
    var stacks = Array.ofDim[scala.collection.mutable.Stack[String]](n_stacks)
    for (i <- 1 to n_stacks) {
      stacks(i-1) = scala.collection.mutable.Stack[String]()
    }

    val instructions = Iterator.continually(scala.io.StdIn.readLine).takeWhile(_ != "quit")
    for (instruction <- instructions) {
      val input = instruction split " "
      val operation = input(0)
      if (operation == "push") {
        val dst   = input(1).toInt
        val block = input(2)
        stacks(dst-1).push(block)
      }
      if (operation == "pop") {
        val src   = input(1).toInt
        val block = stacks(src-1).pop
        println(block)
      }
      if (operation == "move") {
        val src = input(1).toInt
        val dst = input(2).toInt
        stacks(dst-1).push(stacks(src-1).pop)
      }
    }
  }
}
